src/App.tsx

```diff
@@ -1,26 +1,289 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+/* eslint-disable jsx-a11y/label-has-associated-control */
+import React, { useEffect, useState, useRef } from 'react';
 import { UserWarning } from './UserWarning';
+import { USER_ID } from './api/todos';
+import { Todo } from './types/Todo';
+import { createTodos, getTodos, updateTodos, deleteTodos } from './api/todos';
 
-const USER_ID = 0;
+enum Filter {
+  ALL = 'all',
+  ACTIVE = 'active',
+  COMPLETED = 'completed',
+}
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [error, setError] = useState<string | null>(null);
+  const [filter, setFilter] = useState<Filter>(Filter.ALL);
+  const [loading, setLoading] = useState<boolean>(false);
+  const [newTodoTitle, setNewTodoTitle] = useState('');
+  const [loadingTodoId, setLoadingTodoId] = useState<number | null>(null);
+
+  const [counterOfActiveTodos, setCounterOfActiveTodos] = useState(0);
+
+  useEffect(() => {
+    setLoading(true);
+    getTodos()
+      .then(fetchedTodos => {
+        setTodos(fetchedTodos);
+        // Рассчитываем количество активных задач после обновления состояния
+        setCounterOfActiveTodos(
+          fetchedTodos.filter(todo => !todo.completed).length,
+        );
+      })
+      .catch(err => {
+        setError('Unable to load todos' + err.message);
+      })
+      .finally(() => setLoading(false));
+  }, []);
+
+  useEffect(() => {
+    if (error) {
+      const timer = setTimeout(() => setError(null), 3000);
+
+      return () => clearTimeout(timer);
+    }
+  }, [error]);
+
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    if (inputRef.current) {
+      inputRef.current.focus();
+    }
+  }, [todos]);
+
+  function addTodo(title: string) {
+    const trimmedTitle = title.trim(); // Обрезаем пробелы
+
+    if (trimmedTitle.length === 0) {
+      setError('Title should not be empty');
+
+      return;
+    }
+
+    const tempTodo = {
+      title: trimmedTitle, // Используем обрезанное название
+      userId: USER_ID,
+      completed: false,
+      id: Date.now(), // Временный ID
+    };
+
+    setTodos(currentTodos => [...currentTodos, tempTodo]); // Добавляем временную задачу
```

Comment:
Remove comments. Try to write obvious code that explains all we need without comments)